lib/Backends/CPU/FunctionSpecializer.cpp

```diff
@@ -343,7 +339,34 @@ class FunctionSpecializer {
 } // namespace
 
 void LLVMIRGen::performSpecialization() {
-  FunctionSpecializer FuncSpecializer(llmodule_->getFunction("main"));
+  // Specialize all invocations of main functions with constant parameters to
+  // produce a better optimized code.
+  auto *func = getModule().getFunction("jitmain");
+  while (func) {
+    // Call of a "main" function is always the last instruction before return.
+    auto instrBeforeTerminator =
+        std::prev(func->back().getTerminator()->getIterator());
+    auto *call = dyn_cast<llvm::CallInst>(instrBeforeTerminator);
+    if (!call)
+      break;
+    // Specialize only calls of all "main" functions.
+    func = call->getCalledFunction();
+    if (!func || !func->getName().startswith("main"))
+      break;
+    call = specializeCallWithConstantArguments(call);
```

Comment:
Out-of-curiosity why do we only specialize main and not all functions?
Also, why do we do this in glow and not in LLVM proper so that every LLVM users can benefit from it?